compiler/rustc_codegen_llvm/src/llvm/enzyme_ffi.rs

```diff
@@ -1,28 +1,28 @@
-#![allow(non_camel_case_types)]
+#![allow(non_camel_case_types, dead_code)]
```

Comment:
Note to reviewers: seems like this is new code, but a bunch of it is dead. Should I also clean this file, or are there plans to expand uses in the future?

Reply:
Please leave it in place for now as @ZuseZ4 is working on completing this feature and it has been split over multiple PRs due to being a huge lift.